embulk-core/src/main/java/org/embulk/exec/PreviewExecutor.java

```diff
@@ -174,10 +176,15 @@ public void add(Page page)
         @Override
         public void finish()
         {
+            if (res != null) {
+                // if PreviewResult object already exists, we should reuse it and throw PreviewedNoticeError.
```

Comment:
The previous `PreviewResult` is reused in the current fix. We also can add logging "Ignore that the finish method is called twice in preview" for plugin developers. How about this?